plugins/catalog-backend/src/ingestion/processors/BuiltinKindsEntityProcessor.ts

```diff
@@ -193,6 +194,29 @@ export class BuiltinKindsEntityProcessor implements CatalogProcessor {
         RELATION_PART_OF,
         RELATION_HAS_PART,
       );
+
+      // TODO: Consider building a helper around this, but it's a bit difficult
+      // with "deleting" the previous field.
+
+      // TODO: Emitting the attachment from the definition field for now. That
+      // still requires that the definition is filled with the full data and we
+      // we can do any special caching around it for now. Later we can read it
+      // form a definitionLocation field instead.
+      const key = 'definition';
+      // TODO: Also support directly reading from a location in the future, this
+      // avoid that we bload our memory with all definitions.
+      const data = Buffer.from(api.spec.definition!, 'utf8');
+      // TODO: Actually a hard question, depends either on the use case or data,
+      // but for api definitions plain should be a fine choice.
+      const contentType = 'text/plain';
+      emit(result.attachment('definition', data, contentType));
```

Comment:
Assuming you wanted to use `key` here too. I think the format of the key is a bit of an interesting discussion too, whether they're a bit more like relations in that they don't have too much namespacing, or more like annotations to avoid conflicts, group them, versioning, etc. Thinking about this as `definition` vs `backstage.io/api-definition`. Also ofc the scope, are these keys per kind or global? I could see something like an avatar attachment being quite interesting as something you could potentially put on any entity for quicker discovery.

Reply:
For now I thought about having them kind specific, so namespacing would be required. But internally, we would already have a case that is integration specific - so global names with namespacing would be interesting. Luckily Fredrik already suggested to encode the url parameters, so including a slash would be fine - in the end this is about documentation, but one part of this PR should be a small section in the docs. I will make sure to address it there.